logging/config.go

```diff
@@ -140,25 +141,24 @@ func NewConfigFromMap(data map[string]string, components ...string) (*Config, er
 	}
 
 	lc.LoggingLevel = make(map[string]zapcore.Level)
-	for _, component := range components {
-		if ll := data["loglevel."+component]; len(ll) > 0 {
-			level, err := levelFromString(ll)
-			if err != nil {
-				return nil, err
+	for k, v := range data {
+		if component := strings.TrimPrefix(k, "loglevel."); component != k {
```

Comment:
`&& component != ""` ?

Reply:
TrimPrefix will return the original string unaltered, if the prefix is not present. So this line is removing the prefix if it is there, then seeing if anything changed (showing that the prefix was present).

Reply:
I know, but key be "loglevel.", then removing prefix would be "".

Reply:
Done.